### PR TITLE
[#125] HeaderBar cleanup

### PR DESCRIPTION
**Pull request by jtojnar**
I thought I had already sent a PR.

This PR moves the About and Preferences buttons into a [application menu](https://lazka.github.io/pgi-docs/Gio-2.0/classes/MenuItem.html#Gio.MenuItem). In GTK, the easiest and cleanest way to add a menu is to pack it into a [resource](https://developer.gnome.org/gio/stable/GResource.html#GResource.description) file and load that into the application. As a consequence, this PR also incorporates the existing css into aforementioned resource file.

screenshot from 2017-01-31 10-28-37

Closes: #125

### CODE REVIEW

**Comment by elbenfreund** [comment]
I think it would be good to elaborate a bit on the use of GTK resource

File: hamster_gtk/hamster_gtk.py [hunk 1]
```diff
@@ -198,6 +164,24 @@ def save_config(self, config):
         config_helpers.write_config_file(cp_instance, 'hamster-gtk', 'hamster-gtk.conf')
         self.controller.signal_handler.emit('config-changed')
 
+    def _create_actions(self):
+        """Create actions and add register them in the application."""
```
Comment (elbenfreund) [inline hamster_gtk/hamster_gtk.py]: remove 'add'

File: hamster_gtk/resources/gtk/menus.ui [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<interface>
```
Comment (elbenfreund) [inline hamster_gtk/resources/gtk/menus.ui:1]: Should we specify XML standard version and encoding?

**Comment by jtojnar** [comment]
Updated. Now, thinking about it, compiling resources should be incorporated into the build/packaging process. Where exactly should I add it? As a dependency to some Makefile rule?

**Review by elbenfreund — CHANGES_REQUESTED** [review]

**Comment by elbenfreund** [comment]
We probably also need to make sure that resources are build as part of the release process /dist target as well as that the resulting resource file is included in the package.

File: hamster_gtk/resources/gtk/menus.ui (hunk 2 above)
Comment (jtojnar) [reply hamster_gtk/resources/gtk/menus.ui:1]: For XML 1.0, the header is optional.  It defaults to some Unicode encoding.

**Comment by jtojnar** [comment]
I still do not understand the build process. Do I modify Makefile, MANIFEST.in or setup.py?

**Comment by elbenfreund** [comment]
All of the above. :)
1. We need to declare ``build-resources`` a dependency for the dist target.
2. We need to [include the resulting resource file](http://python-packaging-user-guide.readthedocs.io/distributing/#package-data) as part of the packaging in ``setup.py``
3. We most likely need to inform our manifest.in about the new file so it doesn't issue warning.

If you prefer I could address those packaging related issues though...
Also, it may be better to not just dump the resources file in the main python code dir but use a dedicated build dir similar to what we do in ``hamster-shell-extension``.

**Comment by jtojnar** [comment]
If you could tackle it, that would be great. I only see the extension creating a zipball:

https://github.com/projecthamster/hamster-shell-extension/blob/master/build.sh#L3

**Comment by elbenfreund** [comment]
done. Some tests seem still trivially broken tough...

File: setup.py [hunk 3]
```diff
@@ -48,6 +48,6 @@
     ''',
 
     package_data={
-        'hamster_gtk': ('resources/hamster-gtk.gresource',),
+        'hamster_gtk': ['resources/hamster-gtk.gresource'],
```
Comment (elbenfreund) [inline setup.py]: What the reason for  using a ``list`` instead of a ``tuple``?

Comment (jtojnar) [reply setup.py]: sdist is concatenating some list with the `package_data` items so with tuple it fails on the following error:

```
TypeError: can only concatenate list (not "tuple") to list
```

**Review by elbenfreund — APPROVED** [review]

**Pull request merged**